framework/src/tinyalsa/tinyalsa.c

```diff
@@ -673,6 +683,14 @@ int pcm_readi(struct pcm *pcm, void *data, unsigned int frame_count)
 					}
 				}
 
+			} else if (msg.msgId == AUDIO_MSG_XRUN) {
+				/* Handle underrun */
+				/* Try to restart the codec and continue reading data */
+				if (!pcm->draining) {
+					pcm_stop(pcm);
```

Comment:
In my opinion, need to check for error cases.

Reply:
The error is XRUN. That is what we check with "else if (msg.msgId == AUDIO_MSG_XRUN)".
There is no sub-error codes withing XRUN.